be/src/olap/tablet.cpp

```diff
@@ -1772,11 +1796,31 @@ Status check_version_continuity(const std::vector<RowsetMetaSharedPtr>& rs_metas
     return Status::OK();
 }
 
-Status Tablet::write_cooldown_meta() {
+// It's guaranteed the write cooldown meta task would be invoked at the end unless BE crashes
+// one tablet would at most have one async task to be done
+void Tablet::async_write_cooldown_meta() {
+    WriteCooldownMetaExecutors::GetInstance()->submit(tablet_id(), [this]() {
```

Comment:
`this` seems unsafe

Reply:
Seems that it's not a good idea to provide `Tablet::async_write_cooldown_meta`, just make a closure and capture tablet's copied shared ptr, call `tablet->write_cooldown_meta` in this closure.

Reply:
If `write_cooldown_meta` returns not ok and not aborted, enqueue this tablet to executor again.